pkg/infradriver/ssh_infradriver.go

```diff
@@ -1,56 +1,143 @@
-// alibaba-inc.com Inc.
-// Copyright (c) 2004-2022 All Rights Reserved.
+// Copyright © 2022 Alibaba Group Holding Ltd.
 //
-// @Author : huaiyou.cyz
-// @Time : 2022/8/7 5:44 PM
-// @File : ssh_infradriver
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
 //
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 package infradriver
 
 import (
+	"fmt"
+	"github.com/imdario/mergo"
 	v1 "github.com/sealerio/sealer/types/api/v1"
+	v2 "github.com/sealerio/sealer/types/api/v2"
+	"github.com/sealerio/sealer/utils/ssh"
 	"net"
 )
 
-// TODO, implement all functions
+type SSHInfraDriver struct {
+	sshConfigs   map[string]ssh.Interface
+	hosts        []net.IP
+	roleHostsMap map[string][]net.IP
+}
 
-func (d *SSHInfraDriver) Copy(host net.IP, srcFilePath, dstFilePath string) error {
+func NewInfraDriver(cluster *v2.Cluster) (InfraDriver, error) {
+	var err error
+	ret := &SSHInfraDriver{
+		sshConfigs:   map[string]ssh.Interface{},
+		hosts:        cluster.GetAllIPList(),
+		roleHostsMap: map[string][]net.IP{},
+	}
+
+	// initialize sshConfigs field
+	for _, host := range cluster.Spec.Hosts {
+		if err = mergo.Merge(&host.SSH, &cluster.Spec.SSH); err != nil {
+			return nil, err
+		}
+		for _, ip := range host.IPS {
+			ret.sshConfigs[ip.String()] = ssh.NewSSHClient(&host.SSH, true)
+		}
+	}
+
+	// initialize roleHostsMap field
+	for _, host := range cluster.Spec.Hosts {
+		for _, role := range host.Roles {
+			ret.roleHostsMap[role] = host.IPS
```

Comment:
need append, because different host maybe have same role.

Reply:
Any update？

Reply:
it already fixed, pls review.